### PR TITLE
Rename `sgn` with `sign`

### LINKED ISSUE #162: Implement `numpy.sign` equivalent
### Description

This seems to be missing: https://numpy.org/doc/stable/reference/generated/numpy.sign.html

Something like this should work?

```python
def sign(x):
  return pt.switch(pt.eq(x, 0), 0, pt.switch(x > 0, 1, -1))
```

### PR DESCRIPTION
**Pull request by sudarsan2k5**
fixes #162
- Rename sgn with sign
- Deprecated sgn

### Checklist
+ [x] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [x] Link relevant issues, preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+ [x] The commits correspond to [_relevant logical changes_](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes). Note that if they don't, we will [rewrite/rebase/squash the git history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_rewriting_history) before merging.
+ [x] Are the changes covered by tests and docstrings?

### CODE REVIEW

**Review by twiecki — APPROVED** [review]

**Comment by twiecki** [comment]
@sudarsan2k5 Thanks! Note the conflicts, you need to rebase from main.

**Comment by sudarsan2k5** [comment]
by mistakenly converted to draft

File: pytensor/tensor/math.py [hunk 1]
```diff
@@ -1092,7 +1092,7 @@ def log1p(a):
 
 
 @scalar_elemwise
-def sgn(a):
+def sign(a):
     """sign of a"""
```
Comment (ricardoV94) [inline pytensor/tensor/math.py:1098]: Do you mind adding a compatible `sgn` like this (and make it accessible from `pytensor.tensor` as before.

```suggestion
def sgn(a):
    warnings.warn(
        "sgn is deprecated and will stop working in the future, use sign instead.", 
        FutureWarning,
    )
    return sign(a)
```

Comment (sudarsan2k5) [reply pytensor/tensor/math.py:1098]: Hi @ricardoV94 I have implemented the suggested changes, could you please check if we need anything else in this PR?

**Comment by ricardoV94** [comment]
@sudarsan2k5 Sorry I just wanted to check locally the old import is still working. I'll do it tomorrow and merge after

**Comment by ricardoV94** [comment]
Thanks @sudarsan2k5, merged!

**Pull request merged**